### ISSUE TITLE
Centralize input events

### ISSUE THREAD

**Issue by ngokevin** [aframevr/aframe#616]
<a href="https://github.com/dmarcos"><img src="https://avatars.githubusercontent.com/u/39342?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [dmarcos](https://github.com/dmarcos)**
_Thursday Nov 12, 2015 at 19:26 GMT_
_Originally opened as https://github.com/aframevr/aframe-core/issues/409_

---

We need something like an `input-manager`. This would be a singleton providing a global service to get high level events coming from multiple input sources like `keypressed`, `buttonpressed`, `mousemoved`.  This service might be exposed through the scene so components or user scripts can listen the emitted events

**Comment by ngokevin** [comment]
<a href="https://github.com/caseyyee"><img src="https://avatars.githubusercontent.com/u/1560884?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [caseyyee](https://github.com/caseyyee)**
_Thursday Nov 12, 2015 at 22:40 GMT_

----

Kind of what the web API's already do for us no?   A bit onerous IMO.

I was thinking more of the unity input manager approach:
http://docs.unity3d.com/Manual/class-InputManager.html

Basically takes hard inputs and maps them to named controls that you use within your game/app.

**Comment by ngokevin** [comment]
Our events bubble, I think that is sufficient. Individual control components handle the mapping at the moment
